x/programs/cmd/simulator/cmd/key.go

```diff
@@ -6,72 +6,49 @@ package cmd
 import (
 	"context"
 	"fmt"
+	"time"
 
-	"github.com/spf13/cobra"
+	"github.com/akamensky/argparse"
+	"go.uber.org/zap"
 
 	"github.com/ava-labs/avalanchego/utils/logging"
 	"github.com/ava-labs/hypersdk/crypto/ed25519"
 	"github.com/ava-labs/hypersdk/state"
-
 	"github.com/ava-labs/hypersdk/x/programs/cmd/simulator/vm/storage"
 )
 
-func newKeyCmd(log logging.Logger, db *state.SimpleMutable) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "key",
-		Short: "Manage private keys",
-	}
-	cmd.AddCommand(
-		newKeyCreateCmd(log, db),
-	)
-	return cmd
-}
+var _ Cmd = (*keyCreateCmd)(nil)
 
 type keyCreateCmd struct {
+	cmd *argparse.Command
+
 	log  logging.Logger
-	db   *state.SimpleMutable
-	name string
+	db   **state.SimpleMutable
+	name *string
 }
 
-func newKeyCreateCmd(log logging.Logger, db *state.SimpleMutable) *cobra.Command {
-	c := &keyCreateCmd{
-		log: log,
-		db:  db,
-	}
+func (c *keyCreateCmd) New(parser *argparse.Parser, db **state.SimpleMutable) {
```

Comment:
This double pointer `**` is used because the `Simulator.Init` is called after the subcommands have been generated, and the db has to be passed before that the command line arguments have been parsed. Kind of a chicken-egg proble; but this is how I overcome it, if you have a better-looking solution I'm all ears!

Reply:
It looks like you could just move s.Init up in the function.

Reply:
Looks like the db is used in most of the commands, could just add it to the Cmd.Run() signature as a parameter

Reply:
Is the parser also required to be run before the simulator inits?  I'm assuming most initialization isn't dependent on the args anyway, so it seems like you could run the init before you create the parsing commands.

Reply:
Yes, there is the `logLevel` flag which is ran in the initialization and has to be parsed from the args, I will pass the db to the `Run` command though, that makes much more sense

Reply:
Yep passing the `db` to the `Run` func is much better. Thanks !